examples/flutter_gallery/lib/demo/expansion_panels_demo.dart

```diff
@@ -185,27 +189,30 @@ class _ExpansionPanelsDemoState extends State<ExpasionPanelsDemo> {
         builder: (DemoItem<String> item) { // ignore: argument_type_not_assignable, https://github.com/flutter/flutter/issues/5771
           void close() {
             setState(() {
+              // Clear the input state.
+              // TODO(mpcomplete): is there a way to have a Form.reset that purges its children?
+              inputKey = new UniqueKey();
               item.isExpanded = false;
             });
           }
 
-          return new CollapsibleBody(
-            margin: const EdgeInsets.symmetric(horizontal: 16.0),
-            child: new Form(
+          return new Form(
+            key: form1Key,
+            child: new CollapsibleBody(
+              margin: const EdgeInsets.symmetric(horizontal: 16.0),
               child: new Padding(
                 padding: const EdgeInsets.symmetric(horizontal: 16.0),
-                child: new Input(
+                child: new InputFormField(
+                  key: inputKey,
                   hintText: item.hint,
                   labelText: item.name,
-                  value: new InputValue(text: item.value),
-                  formField: new FormField<String>(
-                    setter: (String val) { item.value = val; }
-                  ),
+                  initialValue: new InputValue(text: item.value),
+                  onSaved: (InputValue val) { item.value = val.text; },
                 ),
               ),
-            ),
-            onSave: close,
-            onCancel: close
+              onSave: () { form1Key.currentState.save(); close(); },
+              onCancel: close
+            )
```

Comment:
btw, minor side-note, please move the onSave and onCancel arguments to above the child argument, so that it's clearer which widget it belongs to. Both Adam and I independently misread this the first time (I thought they were on Form, he thought they were on InputFormField).